src/sink/wavefront.rs

```diff
@@ -151,10 +173,15 @@ impl<'a> Pad<'a> {
         }
     }
 
-    pub fn is_pad_zero(&self) -> bool {
+    pub fn skip_pad(&self, pad_control: &PadControl) -> bool {
         match *self {
             Pad::Zero(_, _) => true,
-            Pad::Telem(_) => false,
+            Pad::Telem(x) => match x.kind() {
+                AggregationMethod::Histogram => !pad_control.histogram,
```

Comment:
Why are these being negated?

Reply:
The PadControl structure is affirmative, if the value is true then we pad. Here we're deciding whether to skip adding a pad, so we need the opposite truth value. That make sense?

Reply:
Got it.